domain/secretbackend/state/state.go

```diff
@@ -164,7 +287,7 @@ ORDER BY b.name`, SecretBackendRow{})
 	return rows.toSecretBackends(), errors.Trace(err)
 }
 
-func (s *State) getSecretBackend(ctx context.Context, columName string, v string) (*coresecrets.SecretBackend, error) {
+func (s *State) getSecretBackend(ctx context.Context, tx *sqlair.TX, columName string, v string) (*secretbackend.SecretBackend, error) {
```

Comment:
I missed this. You are not using this `tx`, so it is not running in the transactions that you pass to it.

Reply:
Ops, thanks! I'm fixing this in https://github.com/juju/juju/pull/17064.